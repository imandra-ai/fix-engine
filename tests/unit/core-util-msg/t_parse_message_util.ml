(* a list of messages to test *)
let test_data_l : string list =
  Test_engine_data.fix_data |> String.split_on_char '\n' |> List.map String.trim
  |> List.filter (fun s -> s <> "")

let test_data : string = test_data_l |> String.concat ""

let () =
  let seq = Parse_message_util.split_into_key_value '|' test_data in
  Printf.printf "number of k/v pairs: %d\n%!" (CCSeq.length seq)

let () =
  let seq =
    Parse_message_util.split_into_key_value '|' test_data
    |> Parse_message_util.split_into_messages
  in
  Printf.printf "number of messages: %d\n%!" (CCSeq.length seq);
  assert (CCSeq.length seq = List.length test_data_l)

let () =
  let seq =
    Parse_message_util.split_into_key_value '|' test_data
    |> Parse_message_util.split_into_messages
  in
  let msg0 = CCSeq.take 1 seq |> CCSeq.to_list |> CCList.hd in
  Printf.printf "first message: %s\n%!" (Message.show msg0)

(* test long sequence of messages *)
let () =
  (* repeat [test_data] many times *)
  let n_copies = 1000 in
  let expected_len = n_copies * List.length test_data_l in
  let data = String.concat "" @@ CCList.repeat n_copies test_data_l in

  let seq =
    Parse_message_util.split_into_key_value '|' data
    |> Parse_message_util.split_into_messages
  in
  Printf.printf "number of messages (long): %d\n%!" (CCSeq.length seq);
  assert (CCSeq.length seq = expected_len)
