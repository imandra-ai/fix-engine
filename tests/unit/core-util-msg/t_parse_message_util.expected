number of k/v pairs: 566
number of messages: 15
first message: [("8", "FIX.4.4"); ("9", "399"); ("35", "8"); ("34", "595");
  ("49", "QA_ORDER_GB"); ("56", "QA_00000648_0");
  ("52", "20220912-07:00:24.674988"); ("150", "0"); ("39", "0");
  ("37", "2575671"); ("17", "ZND257567229CQBLL2wO-1");
  ("11", "Tcq.0000.VAS_3e16sWJ"); ("453", "3"); ("448", "934");
  ("452", "122"); ("447", "P"); ("2376", "24"); ("448", "1"); ("452", "3");
  ("447", "P"); ("448", "123"); ("452", "12"); ("447", "P"); ("2376", "22");
  ("581", "3"); ("528", "P"); ("18", "M"); ("9020", "1");
  ("9952", "Rpb_two_simple_valid_co_orders_f"); ("1724", "0"); ("54", "1");
  ("6", "0"); ("14", "0"); ("151", "46"); ("38", "46"); ("40", "P");
  ("59", "0"); ("15", "CHF"); ("207", "XSWX"); ("55", "CH0030170408");
  ("60", "20220912-07:00:24.674998"); ("10", "156")]
number of messages (long): 15000
